app/models/message.rb

```diff
@@ -94,4 +100,19 @@ def no_push_necessary?(sub)
     membership = sub.user.chat_channel_memberships.order("last_opened_at DESC").first
     membership.last_opened_at > 40.seconds.ago
   end
+
+  def rich_link_article(a)
+    if a["href"].include?("//#{ApplicationConfig['APP_DOMAIN']}/") && a["href"].split("/")[4]
+      Article.find_by_slug(a["href"].split("/")[4].split("?")[0])
+    end
+  end
+
+  def send_email_if_appropriate
+    return if chat_channel.channel_type != "direct" ||
```

Comment:
I noticed some occurences of the "direct" value checking for `channel_type`. Do you think it would be good to add a `direct?` method in the `ChannelType` class, like the `open?` one?

Reply:
Yeah, good call.